mailpoet/assets/js/src/automation/listing/index.tsx

```diff
@@ -1,76 +1,180 @@
 import { Search, TableCard } from '@woocommerce/components/build';
-import { Button, Dropdown, MenuGroup, MenuItem } from '@wordpress/components';
+import { TabPanel } from '@wordpress/components';
 import { __ } from '@wordpress/i18n';
+import { useCallback, useLayoutEffect, useMemo } from 'react';
+import { useHistory, useLocation } from 'react-router-dom';
 import { getRow } from './get-row';
-import { Workflow } from './workflow';
+import { Workflow, WorkflowStatus } from './workflow';
 
 type Props = {
   workflows: Workflow[];
   loading: boolean;
 };
 
 export function AutomationListing({ workflows, loading }: Props): JSX.Element {
-  const rows = workflows.map((workflow) => getRow(workflow));
+  const history = useHistory();
+  const location = useLocation();
+  const pageSearch = useMemo(
+    () => new URLSearchParams(location.search),
+    [location],
+  );
+
+  const updateUrlSearchString = useCallback(
+    (keysAndValues: Record<string, string>) => {
+      Object.keys(keysAndValues).forEach((key) => {
+        pageSearch.set(key, keysAndValues[key]);
+        if (['per_page', 'status'].includes(key)) {
+          pageSearch.delete('paged');
+        }
+      });
+      history.push({ search: pageSearch.toString() });
```

Comment:
This is the one thing I'm still really struggling with. I'm not sure how to update the current tab in the TabPanel's state when a user hits back/forward. Do you have any ideas for how to handle this so users can actually navigate forwards/backwards through the tabs?

Reply:
Thank you for the `key` suggestion! Not ideal, but better than having something non-functional.

> On a different note, I'm not sure why do you check also for the status part in this case (['per_page', 'status'].includes(key)), those params seem unrelated, aren't they?

The idea is to reset to the first page when a user clicks on a different tab or changes the number of results per page. If you're viewing page 2 of "All" and then click on "Inactive" you wouldn't necessarily want to immediately view page 2 of the "Inactive" tab. And if there aren't enough "Inactive" workflows for there to be a second page, the pagination controls won't even appear and you can end up seeing something like this:

<img width="1355" alt="image" src="https://user-images.githubusercontent.com/8656640/188716990-2b1daf3a-b2f6-46d8-84e9-2715ca76300e.png">

It's similar but different for `per_page`. Let's say you have 240 workflows and you're viewing page 5/10 with 25 workflows per page. If you switch to 100 per page there are now only 3 total pages, so leaving `paged=5` would result in an empty table. 

We could try to dynamically calculate a logical new page number. For example we could figure out what the first workflow was on the page you were viewing previously and then switch to whichever page had that workflow with the new `per_page` value. If you were previously looking at workflows 101-125 then we could figure out we should show you 101-200 with the new `per_page`. That just seemed overly complicated and may not even be what users expect.